stl/inc/random

```diff
@@ -2936,7 +2936,7 @@ private:
                 _Vx1 = 2 * _NRAND(_Eng, _Ty) - 1;
                 _Vx2 = 2 * _NRAND(_Eng, _Ty) - 1;
                 _Sx  = _Vx1 * _Vx1 + _Vx2 * _Vx2;
-                if (_Sx < 1) {
+                if (_Sx < 1 && _Sx != 0) {

```

Comment:
> * For `fisher_f_distribution`, should we also reject here if `_Px == _Ty{1}`? That would generate infinity rather than NaN.
> * The issue mentions `student_t_distribution`, but there's no change to it. It's not obvious to me how it would cause NaN, but `_Rs == 0` would give infinity.

That's right. Thank you for that. I just tested the code for isNan(x) and didn't catch isInf(x). I thought `_Px == _Ty{1}` could not happen at all. I fix it now. Thanks.

Here is my test program with a slight improvement of the @statementreply test code on the issue description. however I'm not sure is it a good idea to add time-consuming test to unit tests?

```cpp
#include <cassert>
#include <cmath>
#include <random>
#include <iostream>

#include "\tests\std\tests\GH_001017_discrete_distribution_out_of_range\bad_random_engine.hpp"

using namespace std;

template <class Distribution>
void test(Distribution&& distribution) {	
	for (bad_random_generator rng; !rng.has_cycled_through();) {
		const auto x = distribution(rng);
		assert(!isnan(x) && !isinf(x));
	}
}

int main() {
	test(normal_distribution<>{});
	std::cout << "normal_distribution\t\tok" << std::endl;

	test(lognormal_distribution<>{});	
	std::cout << "lognormal_distribution\t\tok" << std::endl;

	test(fisher_f_distribution<>{});	
	std::cout << "fisher_f_distribution\t\tok" << std::endl;

	test(student_t_distribution<>{});	
	std::cout << "student_t_distribution\t\tok" << std::endl;

	std::cout << "Done!";
	std::cin.get();
	return 0;
}
```

Reply:
> This is fine as far as preventing NaN, but will overflow for `_Sx` less than ~7.8e-306 for `double`. On the other hand, even with `_Vx1` and `_Vx2` on the order of `numeric_limits<double>::min`, the true value of `_Fx` is around 38. So maybe this could be made to work for all nonzero `_Sx`, though it's a bit beyond the scope of the original issue.

Thank you. I've found the exact min value for _Sx that causes the calculation to overflow.